test-app/src/main/java/org/readium/r2/testapp/MediaService.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2022 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by the BSD-style license
+ * available in the top-level LICENSE file of the project.
+ */
+
+package org.readium.r2.testapp
+
+import android.app.PendingIntent
+import android.content.Intent
+import android.os.Build
+import android.os.IBinder
+import androidx.lifecycle.lifecycleScope
+import androidx.media2.session.MediaSession
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.*
+import org.readium.r2.navigator.ExperimentalAudiobook
+import org.readium.r2.navigator.media2.MediaSessionNavigator
+import org.readium.r2.testapp.bookshelf.BookRepository
+import org.readium.r2.testapp.db.BookDatabase
+import org.readium.r2.testapp.reader.ReaderActivityContract
+import org.readium.r2.testapp.utils.LifecycleMediaSessionService
+import timber.log.Timber
+import kotlin.time.ExperimentalTime
+
+@OptIn(ExperimentalTime::class, ExperimentalAudiobook::class, ExperimentalCoroutinesApi::class)
+class MediaService : LifecycleMediaSessionService() {
```

Comment:
This code might be a bit dense for newcomers, could you add some comments to describe it? That's important if they copy-paste it in their apps.

Reply:
Which code?

Reply:
* The purpose of `createSessionActivityIntent()`.
* Why we save the `currentLocator` in the DB in the service instead of the `Activity`.
* Why we close the navigator in `onTaskRemoved()`.

I know implementers can RTFM but since this service is likely to be copy-pasted, I think it could help to have a bit more explanations for people to understand what they have in their apps.

Reply:
I had to read the fucking manual several times and do many experiments to get this stuff working.
Done.